simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp

```diff
@@ -1145,15 +1327,27 @@ geometry_msgs::msg::PoseStamped HdMapUtils::toMapPose(
 }
 
 geometry_msgs::msg::PoseStamped HdMapUtils::toMapPose(
-  traffic_simulator_msgs::msg::LaneletPose lanelet_pose)
+  const traffic_simulator_msgs::msg::LaneletPose & lanelet_pose) const
 {
-  return toMapPose(
-    lanelet_pose.lanelet_id, lanelet_pose.s, lanelet_pose.offset,
-    quaternion_operation::convertEulerAngleToQuaternion(lanelet_pose.rpy));
+  if (const auto pose = clampLaneletPose(lanelet_pose)) {
+    return toMapPose(
+      pose->lanelet_id, pose->s, pose->offset,
+      quaternion_operation::convertEulerAngleToQuaternion(pose->rpy));
+  } else {
+#ifdef ROS_DISTRO_GALACTIC
+    THROW_SEMANTIC_ERROR(
+      "Lanelet pose\n", rosidl_generator_traits::to_yaml(lanelet_pose),
+      "\nis invalid, please check lanelet length and connection.");
+#else
+    THROW_SEMANTIC_ERROR(
+      "Lanelet pose\n", traffic_simulator_msgs::msg::to_yaml(lanelet_pose),
+      "\nis invalid, please check lanelet length and connection.");
+#endif
```

Comment:
Exception messages should not contain newline characters. You might expect this message to be written verbatim to standard output, but there is no guarantee that the exception-catching process will do so. This exception is actually written to the simulation results file instead of standard output. And the messages in the simulation result file are used to display the Evaluator. The Evaluator's simulation results display expects messages to not contain newline characters.

Reply:
to_yaml functions output contains newline characters, so I will remove using to_yaml function in exception message.
If I remove to_yaml function, [this section](https://github.com/tier4/scenario_simulator_v2/blob/d47626c8afbe1c554a74e48a44a28652bfe21402/simulation/traffic_simulator/CMakeLists.txt#L16-L25) will be unnecessary, so I remove it.